### PR TITLE
Migrate PendingVerification in pallet_registrar from value to map

### PR DESCRIPTION
**Pull request by tmpolaczyk**
This allows an unlimited number of para ids to be waiting for verification, by using an unbounded `StorageMap<ParaId, ()>` instead of a bounded `StorageValue<Vec<ParaId>>`.

Breaking change, includes a migration.

### CODE REVIEW

File: runtime/common/src/migrations.rs [hunk 1]
```diff
@@ -377,6 +378,40 @@ where
     }
 }
 
+pub struct RegistrarPendingVerificationValueToMap<T>(pub PhantomData<T>);
```
Comment (girazoki) [inline runtime/common/src/migrations.rs:381]: Can we add a test for the migration?

**Review by girazoki — COMMENTED** [review]
The PR looks good but I would prefer if the migration was tested. Also we can use the pre and post upgrade hooks for this

**Comment by tmpolaczyk** [comment]
> Also we can use the pre and post upgrade hooks for this

The problem is that this storage item can be empty if there are no pending chains, so I don't know what to test for. But I can add an integration test similar to `test_migration_config_full_rotation_period` to check that the key was calculated correctly.

**Comment by girazoki** [comment]
> > Also we can use the pre and post upgrade hooks for this
> 
> The problem is that this storage item can be empty if there are no pending chains, so I don't know what to test for. But I can add an integration test similar to `test_migration_config_full_rotation_period` to check that the key was calculated correctly.

So I think:

- You manually insert in this storage item certain chains with the previous layout.
- You run on_runtime_upgrade.
- You check the new storage map is populated correctly

Yes similar to `test_migration_config_full_rotation_period`

**Comment by tmpolaczyk** [comment]
Yes, that's already done and it works :)

**Review by girazoki — APPROVED** [review]

**Review by nanocryk — APPROVED** [review]

**Pull request merged**